java/org/puredata/core/NativeLoader.java

```diff
@@ -48,13 +48,20 @@ public NativeLibraryLoadError(String message, Throwable cause) {
     detectSystem();
   }
 
-  private static void detectSystem() {
-    osArch = System.getProperty("os.arch").toLowerCase();
-    if (osArch.indexOf("64") != -1) {
-      osArch = "x86_64";
-    } else if (osArch.indexOf("86") != -1) {
-      osArch = "x86";
+  private static String determineOsArch(){
+    final String arch = System.getProperty("os.arch").toLowerCase();
+    if (arch.indexOf("aarch64") != -1) return "arm64";
```

Comment:
Can you keep the same formatting as the previous checks?

```java
if (arch.indexOf("aarch64") != -1) {
  return "arm64";
}
```

Reply:
Seems sensible. Also removed the superfluos `else if`.